sbi/inference/potentials/base_potential.py

```diff
@@ -61,3 +64,22 @@ def return_x_o(self) -> Optional[Tensor]:
         `self._x_o` is `None`.
         """
         return self._x_o
+
+
+class CallablePotentialWrapper(BasePotential):
+    """If `potential_fn` is a callable it gets wrapped as this."""
+
+    allow_iid_x = True  # type: ignore
+
+    def __init__(
+        self,
+        callable_potential,
+        prior: Optional[Distribution],
+        x_o: Optional[Tensor] = None,
+        device: str = "cpu",
+    ):
+        super().__init__(prior, x_o, device)
+        self.callable_potential = callable_potential
+
+    def __call__(self, theta, **kwargs):
+        return self.callable_potential(theta=theta, x_o=self.x_o, **kwargs)
```

Comment:
We are trusting the user to pass a `callable_potential` that has `theta` and `x_o` `args, right? 
Or is there any check that is performed above?

Reply:
Added a check.